lib/core/structures/replication-structures.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * A structure describing a replica set configuration that needs to be passed to
+ * MongoDB during replication initiation and reconfiguration
+ */
+class ReplicationConfiguration {
+  constructor(replicaSetName, hostnames, port) {
+    this.replicaSetName = replicaSetName
+    this.hostnames = hostnames
+    this.port = port
+  }
+
+  /**
+   * Returns the configuration in the structure required by the MongoDB commands:
+   * - replSetInitiate
+   * - replSetReconfig
+   */
+  toMongoConfiguration() {
+    const hosts = this.hostnames.map((host, index) => ({ _id: index, host: `${host}:${this.port}` }))
+    const configuration = {
+      _id: this.replicaSetName,
+      members: hosts
+    }
+    if (this.port === 27019) configuration.configsvr = true
```

Comment:
What about being able to inject custom config in the constractor 

So just doing  this 

```js
return {...this.defaultConfiguration, ...configuration}`
```

Reply:
Lets add this, if we ever need it?

Reply:
I changed the constructor to be extendible